app/views/products/new.html.erb

```diff
@@ -0,0 +1,14 @@
+<h1>Add a Product</h1>
+
+<section>
+<%= form_for @product, url: p_create_path(params[:vendor_id]) do |f| %>
```

Comment:
You might be able to simplify this by doing something like `form_for [@vendor, @product] do |f|`. You would need to set `@vendor` in your controller, of course.

Using this version, `form_for` knows that we want a path to create a Product (it knows this because `@product` is a new, unsaved Product instance), and it also knows that it needs to find a route that involves a Vendor.

It'll basically find a route like `POST /vendors/:vendor_id/products`, which is exactly what we want I believe.